iothub/service/src/Messaging/MessagingClient.cs

```diff
@@ -0,0 +1,379 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Amqp;
+using Microsoft.Azure.Amqp.Framing;
+using Microsoft.Azure.Devices.Common;
+using Microsoft.Azure.Devices.Common.Exceptions;
+using Microsoft.Azure.Devices.Http2;
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// Transport types supported by MessagingClient.
+    /// </summary>
+    /// <remarks>
+    /// Amqp and Amqp over WebSocket only.
+    /// </remarks>
+    public enum TransportType
```

Comment:
This enum is applicable to the other subclients you're adding in this PR, so it should live in its own file.